spinn_front_end_common/interface/abstract_spinnaker_base.py

```diff
@@ -440,8 +442,8 @@ def __init__(
         self._mapping_tokens = None
         self._load_outputs = None
         self._load_tokens = None
-        self._last_run_outputs = None
-        self._last_run_tokens = None
+        self._last_run_outputs = list()
+        self._last_run_tokens = list()
```

Comment:
Why do these need to be empty lists?

Reply:
this is a side effect of the get_generated_output. so that you dont need to check for None before asking for in.

Reply:
This also comes from another branch, so removing from here for now...